%
% NOTE -- ONLY EDIT ScISI.Rnw!!!
%
%\VignetteIndexEntry{ScISI Working Paper}
%\VignetteDepends{}
%\VignetteKeywords{Interactome}
%\VignettePackage{ScISI}
\documentclass[11pt]{article}

\usepackage{hyperref}
\usepackage{natbib}
\usepackage{subfigure}
\usepackage{graphicx}
\usepackage{amsmath,pstricks,fullpage,amsthm,amssymb}




\textwidth=6.2in
\textheight=8.5in
%\parskip=.3cm
\oddsidemargin=.1in
\evensidemargin=.1in
\headheight=-.3in

\newcommand{\Rfunction}[1]{{\texttt{#1}}}
\newcommand{\Robject}[1]{{\texttt{#1}}}
\newcommand{\Rpackage}[1]{{\textsf{#1}}}
\newcommand{\Rmethod}[1]{{\texttt{#1}}}
\newcommand{\Rfunarg}[1]{{\texttt{#1}}}
\newcommand{\Rclass}[1]{{\textit{#1}}}



\newcommand{\classdef}[1]{%{\em #1}
}
\newtheorem{Lem}{Lemma}


\begin{document}
\title{Creating in Silico Interactomes}
\maketitle

\section{Introduction}

%%FIXME: we need to describe creating interactomes - not just one
Understanding protein interactions is relevant to our understanding of
complex biological processes and to deciphering the different roles
played by genes and proteins. Since proteins seldom act alone, but
rather tend to form multi-protein complexes which then perform
different molecular tasks it seems that this is an important level of
organization.  In this paper we discuss the problem of estimating the
set of protein complexes that exist in a cell. In particular we
propose a sound statistical approach to this estimation problem.
There are several high through-put experiments that provide data on
protein interactions and hence on the estimation problem we are
interested in. These include the yeast two-hybrid system (Y2H) with
data from \cite{Ito, Uetz, Giot} and high-throughput co-precipitation
experiments such as \citep{Gavin, Ho, Krogan}. 

While several authors have constructed, or estimated, interactomes
by combining these, and other, data our approach is different (FIXME
citations here). Instead, we will
create different interactomes, each relevant to a particular concept
or type of interaction. The resultant 
interactomes can be combined by different users, often according
to different criteria, to yield more comprehensive interactomes. We
believe that there is value in keeping the different concepts separate. 
Binary physical interactions are quite distinct from multi-protein complex 
membership, since the relationships defined by the latter are not binary
and not all members of a protein complex directly interact. Further,
the data sources are quite different, and the experimental errors differ
in type and magnitude according to the assays used. By keeping data types
separate we will be able to better assemble and model the observed data.

Data from high throughput experiments are expensive and time consuming
to carry out. As a result relatively few statistical methods have been
developed and those that have tend to be centered on a particular
experiment and have not been broadly validated.  These observations
suggest that simulation models may be beneficially employed to
identify the viability of experimental procedures, to aid in
integrating data from disparate sources, and to provide a sound basis
for designing experiments. In this paper we outline an approach in the
creation of \textit{in silico interactomes} that would be suitable for
such simulation experiments as well as for a number of other analytic
purposes.

We consider the problem of estimating \textit{in silico} interactomes
(ISI) for \textit{Saccharomyces cerevisiae} and we describe five
different roles where such a computational device can be used.  First,
the ISI provides a tool that can be used to mimic wet-lab experiments;
these \textit{in silico} experiments can help to confirm the viability
of the wet-lab experimental procedures by allowing investigators to
alter experimental conditions (by the modification of both stochastic
and systematic error rates) and then measure how the outputs change as
the error rates do. Second, the ISI provides a tool from which
multiple data sets can be generated, under different conditions, so
that the statistical properties of different proposed estimation
procedures can be directly compared. A third use for the ISI is to
develop tools and strategies for small scale experiments that can
probe the interactome at a very detailed level. For example, a
reasonable strategy to investigate a set of protein complexes is to
first select a small number of candidate genes and explore their
interactions via co-precipitation experiments. Once the first
experiment has been collected one would like to analyze the data and
determine an optimal, or nearly optimal set of baits for future
experiments. A fourth use of an ISI is to study the effect of
perturbations in a network.  Such perturbations could be caused by a
mutation, gene deletion or treatment with a drug.  The fifth use for
such a tool is to explore and understand the effects of different
sampling paradigms on the inferences that can drawn from different
wet-lab experiments. We note that there are a number of recent papers
\citep{Oltavi,Goverview, SubsampNetworks} that raise many important
issues in this regard.

An ISI is not without its limitations. It cannot establish new
relationships between its members, but rather can only reflect the
current state of biological knowledge. It can, however, be used in any
of the five tasks listed above and the outputs of the computational
experiments will often provide predictions of real interactions that
can be tested. If these predictions are accurate there will be a
resultant savings in cost as wet lab experiments can be directly
tailored to the set of likely candidates. Further, one type of
interaction can be used either to predict or to model other types of
interactions and to provide information about the likelihood of
interactions; the verification of interactions, however, remains the
domain of experimental biology.
%%FIXME: Xia's paper could be cited here

Protein interactions are dynamic by nature so that different
interactions occur at different times and under different
conditions. For example, some protein complexes such as those needed
to facilitate cell division are often transient in nature appearing
when needed and dissolving when no longer required, whereas other
complexes such as the ribosome (and its subunits) are essentially
permanently constituted.  We provide some tools that will allow
investigators the ability to indicate situations and conditions under
which an interaction is known to occur and we discuss a paradigm where
an investigator will be able to select all interactions for a given
condition.

%%I if we are going to validate the interactome we are building, should
%%we not regulate the estimate behind the method?? -- TC
While the estimated interactome is itself of some great interest, we
note that it is more important to describe the procedure by which an
ISI is produced. There are many reasons why the estimate is of less
importance, among them the fact that the methodology can be applied to
many organisms and tissue types to yield specialized ISI estimates. We
also note that the data used in the construction is constantly being
updated and improved, hence users will want to regularly update any
estimate as new data sources become available, or old ones are
updated. Hence, to be viable there must be a mechanism for
updating and refining the ISI. We also note that there are many
different and disparate data sources and the system should be
sufficiently flexible to allow users to select data sources
appropriate for their studies as well as the ability to incorporate
additional data from new and potentially different experimental
techniques. Finally we note that investigators will often want to
modify the ISI, either with locally produced, though not yet public,
data, or to satisfy personal beliefs. The paradigm we describe here
allows investigators to create their own personalized estimates, to
share interactomes and to revise and update interactomes.

In this paper we first define the notion of a biological interactome
and give various examples of such interactomes. We then present a set
of computational tools, embodied in the \Rpackage{ScISI} package available
through the Bioconductor project (\url{www.bioconductor.org}), that
we have assembled to construct \textit{in silico interactomes}. The
general steps that must be carried out are, first assembling the data
that are to be used for the construction and second to process those
data into an estimated interactome that can be used as an input for
the various computational tasks described above. Finally we report on
some \textit{in silico} experiments that can be performed using the
estimated interactome.


\section{Preliminaries}

In our discussion we do not distinguish between the term gene and the
term protein, since the available resolution is not fine enough and
equating these concepts leads to a more comprehensible dialog. The
methodology is sufficiently general to be able to address the more
complicated situation of different protein/polypeptide variants should
they arise.

\subsection{Biological Interactomes}

In the broadest sense, an interactome is a set of elements $P$, for
our purposes $P$ is the set of genes, or their related proteins,
within a particular organism or cell type. The elements of $P$ have a
variety relationships with each other and each of these different
relationships can be modeled using some form of graph. Some
relationships are binary, for example, protein $p_1$ is known to
directly physically interact with some other protein, $p_2$, perhaps
under a specific set of conditions. In other cases, the relations are
not binary but rather are one to many or many to many. 
In this setting all members of a complex are related to
each other, but they need not directly interact and the relationship
is not binary. A reasonable model for such data is as either a
hypergraph \cite{Berge} or, equivalently, the bipartite graph
representation. Proteins are grouped into sets according to whether
they are in a complex and these sets constitute the hyperedges of the
hypergraph. There is no need for the hyperedges to be disjoint and
proteins can be in many or few hyperedges, depending on whether the
protein is in many or few complexes.

We can put these concepts into a more biological framework as
follows. Let $P$ denote the set of expressed genes in some cell or tissue,
and hence each such gene is represented by a node in our graph. Any
two genes that have either a direct biochemical or biophysical
interaction will have an edge between them. We remark that a gene may
be associated with itself if it induces a homodimer and such an
observation would be represented by a self-loop in the graph. 
Next, we consider a different set of relationships which are those of
protein complex co-membership. In this, different graph, we again
represent the genes as nodes, but now edges are determined by whether
or not the two proteins are found as  constituent parts of some
protein complex. If represented in this way there is a strict loss of
information, since a graph can only represent binary relationships yet
protein complex membership is a many-to-one relationship. For this
reason we prefer the hypergraph since in this format all important 
information is conserved.

%%FIXME:
%representation: somewhere we probably want to discuss this concept
%There are sure to be situations where a protein complex, that exists
%under one set of conditions, is a proper subset of a complex that
%exists under another set of conditions. For some analyses it will be
%important to remove those complexes which are proper subsets of other
%complexes and we provide tools to help do that.

\begin{figure}
\centering
\subfigure[Protein Binary Interaction Network]{
\includegraphics[width = 65mm]{../extdata/protein_network_large.jpg}}
\caption{A protein binary interaction network as a graph and protein complex
membership network rendered as a bipartite graph FIXME - Cannot use this as it is
probably copyrighted, but wanted to put here as example...need to render one 
ourselves}
\end{figure}


These two biological interactomes are related, but they are also
distinct, and the use of the data in modeling and other downstream
uses requires that some caution be used to not confuse the
concepts. It is certainly the case that for our working definition of
a protein complex the members of that complex will have some binary
interactions with other members. But not all members have direct
physical interactions with other members, so we would not use a
so-called \textit{matrix} model to represent those interactions. These
two examples highlight the subtle differences between two distinct
interactomes and underscores the necessity to keeping distinct
interactomes separate. We will refer to the first of these
interactomes as $I_b$, where the $b$ subscript reinforces the notion
that binary relationships are being modeled and the second as $I_c$
where the $c$ subscript reinforces the notion that protein complexes
are being modeled.

There are of course many limitations to such an approach, however,
there will be ample opportunity to refine and revise the models
proposed here as our knowledge of the underlying biology improves and
as the available computer technology improves. Among the limitations
is the observation that some interactions will only take place under a
specific set of conditions, and that other interactions, sometimes
involving the same proteins, will occur under other conditions. Thus,
the model representing these data will need to become richer and to be
able to indicate under which conditions a given relationship is known
to occur. Some interactions are specific to certain tissues (in higher
organisms) or to particular phases of the cell cycle, and so on. The
models we propose can be extended in fairly straightforward ways to
encompass these details, but the available data are currently to
sparse to warrant such an approach. We will note some experiments, and
some uses of existing experimental data, that can provide some
insight.

In this paper, we will report on a set of methodological tools that
can be used to construct an \textit{in silico interactome} for
\textit{S. cerevisiae}. We restrict our attention to the two
interactomes described above, $I_b$ and $I_c$. As we noted above, not
all complexes are present at all times in the cells under study, but
we are unable to model this aspect due to the lack of appropriate data
on which to base such classifications. When such data become more
widely available they can easily be accommodated within the paradigm
we are proposing.

We first define what we mean by a protein complex and describe
different models for representing these data. Then we explain the
relationship of binary interactions within this working definition of
a protein complex. Next we give a concise description of the
computational tool \textit{ScISI} and show its dynamic properties in
constructing different complex membership interactomes and binary
interactomes. We then describe the process of creating an \textit{in
  silico} protein complex interactome via \textit{ScISI} based on two
public meta-data sources and three experimental data sets by combining
purification of protein complexes with identification of their
individual components via mass spectroscopy. In addition we consider
the construction of an \textit{in silico} binary interactome based
primarily on data from IntAct, which is itself based on high
throughput experimental data assayed by the Yeast 2-Hybrid (Y2H)
technology.  The IntAct repository currently has fourty-one Y2H
data-sets which are parsed and processed in creating the binary
interactome.

%% (FIXME: list the yeast data here) ... can you explain which data you
%% would like listed...certainly not all of the y2h data?  
%% FIXME:RG:March I think we should have a table that lists the main ones - 
%%genome wide ones, with some notion of number of reported baits and number 
%% of reported hits. 

\subsection{Protein complexes: Definition, Representation and
  Characteristics} 

We define a protein complex to be two or more proteins that combine to
form a single, connected functional unit for the purposes of carrying
out some biological objective. There is no assumption that all members
of a complex physically interact with one another, but there must be
sufficient physical binary interactions so that all proteins form a
connected multi-protein structure. In essence, we define a
multi-protein complex as a collection of proteins, and in some cases
polypeptides, that are connected via binary interactions.

\subsubsection{Graphs and HyperGraphs}

We require a small amount of graph theory to allow for succinct
discussion of the relevant concepts. A graph $G$, consists of a pair
of sets; the vertices, or nodes, $V$, and the edges, or relations,
$E$. We write $G=(V,E)$. Graphs can be used to represent binary
relationships and hence are an appropriate model and data structure
for $I_b$. They cannot represent the more complicated situation of
modeling protein complex co-membership. In this case we require the
use of hypergraphs \cite{Berge}. A hypergraph, $H$, consists of a set
of nodes, $V$, and a set of hyperedges. A hyperedge is any subset of
$V$. Thus, for modeling protein interaction data the vertex set is the
set of proteins in the organism or tissue under study. To represent binary
relationships we will use a graph, where the edges indicate the
presence of a binary interaction and for complex co-membership data we
will use a hypergraph, where the hyperedges are the identities of the
proteins that constitute the complex. There is a one-to-one
relationship between hypergraphs and bipartite graphs and we will use these 
terms inter-changeably in the report. Indeed it will be the case that
bipartite graphs are much easier to render pictorially, and so we 
will defer to these graphs rather than the hypergraphs when needed.

There are a variety of different representations for both graphs and
hypergraphs. We will mainly make use of the incidence matrix
representation for $H$ and the adjacency matrix representation of $G$.
For the protein complex hypergraph the
incidence matrix representation, $M_H$, is a $\{0,1\}$-matrix where
proteins index the rows and protein complexes index the columns, with

\begin{equation}
(M_H)_{ij} = 
\begin{cases}
1& \mbox{if protein $i$ is a member of complex $C_j$},\\
0& \mbox{otherwise}.
\end{cases}
\end{equation}


We remark that this representation can be extended to include the
multiplicities of protein membership by simply letting the entries
$(M_H)_{ij} \in \mathbb{N}$. The non-negative integer would imply the
multiplicity by which a protein appears in a given protein complex.

When studying
binary interactions the representation will be of a graph, and then
the adjacency matrix is typically square (or upper triangular) with
the proteins indexing both the rows and the columns,

\begin{equation}
(M_B)_{ij} = 
\begin{cases}
1& \mbox{if protein $i$ directly interacts with protein $j$},\\
0& \text{otherwise}.
\end{cases}
\end{equation}

<<getGraph, echo=FALSE, results = hide>>=
library("ScISI")
library("Rgraphviz")
data(arp23G)
data(nAt)
pdf("arp23Graph.pdf")
plot(arp23G, "dot", nodeAttrs = nAt)
dev.off()
@ 

\begin{figure} 
\centering
\subfigure[Atomic rendering of ARP 2/3 labeled by common names.]{
\label{arp23Physical}
\includegraphics[width = 75mm]{../extdata/Arp2_3_complex.png}}
\subfigure[Graphical rendering of ARP 2/3 identified by crystallography 
labelled by the systematic names.]{
\label{arp23Graph}
\includegraphics[width =75mm]{arp23Graph.pdf}}
\caption{Some scientists might find the physical rendering of ARP 2/3 to be
of more use while others might prefer a graphical representation. We note that
the issue of identifiability presents itself as creator of the atomic rendering
has chosen one version of the common names while the creator of the graph has
chosen the systematic gene names. The coloring scheme provided allows for one
to one matching of proteins from one rendering to the other.}
\end{figure}

\subsubsection{Protein (Complex) Identifiability}

For a protein complex to be uniquely identifiable, it is necessary for each 
of the constituent protein members to be uniquely identifiable. Unfortunately,
this is not always possible. Genes, and their corresponding proteins, are
often referenced using database or investigator specific names and mappings
from one nomenclature to another are not always one to one. We have
used the approach of mapping all identifiers to the set of yeast systematic 
names and all references will be made using systematic names. When the mappings
were not unique we selected the first name in the list.
Table~\ref{geneNames} gives presents some examples of the problems that 
arise when mapping between systematic and common names.

\begin{table}
\centering
\label{geneNames}
\begin{tabular}{|l|p{0.65\textwidth}|}
\hline
Gene Common Names&Corresponding Gene Systematic Names\\
\hline
YRF1&YER190W, YDR545W, YGR296W, YLR466W, YLR467W, YNL339C, YPL283C\\
YPK1&YKL126W, YJL093C, YNL307C\\
POT1&YIL160C, YKL198C\\
ZRG11&YIL046W, YOR030W\\
RTS1&YOR014W\\
\hline
Gene Systematic Names&Corresponding Gene Common Names\\
\hline
YPL084W&NPI3, ASI6, VPS31, LPF2\\
YIL148W& UB11, UBI1, CEP52A\\
YNL041C& SEC37, COD2\\
YGR028W& YTA4\\

YHR133C&NA\\
YPL170W&NA\\
\hline
\end{tabular}
\caption{The correspondence between common names and systematic names in
\textit{S. cerevisiae}. Notice that the correspondence can be many to one in 
both instances and can also be undefined.}

\end{table}

\section {Methodology}

The creation of an interactome is based on the integration of a number
of different data sources. We identify three distinct sources of data
that can be used to construct the protein complex interactome and then
two other sources of data that can be used to create the binary
interactome. We emphasize the construction of the protein complex
interactome as it is quite different from other published protein
interaction data bases, while the binary interaction estimate is quite
similar to other constructs and tends to differ in important, but less
dramatic ways from those estimates.

<<getGraph2, echo=FALSE, results = hide>>=
data(eAt)
pdf("arp23Y2HG2.pdf")
plot(arp23G, "dot", edgeAttrs = eAt, nodeAttrs = nAt)
dev.off()
@ 

<<getGraph3, echo = FALSE, results=hide>>=
data(eAt2)
pdf("arp23InfG.pdf")
plot(arp23G, "dot", edgeAttrs = eAt2, nodeAttrs = nAt)
dev.off()
@ 

\begin{figure} 
\centering
\subfigure[Crystal Graph of Arp 2/3 - the edges represent true protein-protein
interaction derived from crystallography.]{
\label{arp23Crystal}
\includegraphics[width = 50mm]{arp23Graph.pdf}}
\subfigure[Y2H Graph of Arp 2/3 - the directed edges represent affiliation 
from the bait proteins towards the prey proteins.]{
\label{arp23Y2HGraph}
\includegraphics[width =50mm]{arp23Y2HG2.pdf}}
\subfigure[Inferred Protein-Protein Graph of Arp 2/3 - the red edges represent 
computationally inferred interactions.]{
\label{arp23InfGraph}
\includegraphics[width =50mm]{arp23InfG.pdf}}
\caption{These figures show the difference between the true atomic 
structure determined by crystallography and by Y2H experimentation as well 
as crystallography and computationally inferred interactions. The missing 
edges from the Y2H graph are
indeterminate since missing edges can either be due to the relationship
not being tested or due to false negative observations.}
\end{figure}


A fundamental difference needs to be made between the lack of edges
within this graph. An edge is missing between two proteins $p_1$ and
$p_2$ in $I_b$ if either $p_1$ never interacts with $p_2$ (a true
negative interaction) or $p_1$ has never been tested with respect to
$p_2$ (an untested interaction). Problems arise in the structure $I_b$
because neither the bait population nor the prey population is ever
fully documented. In this way, the distinction between true negative
edges and untested edges becomes indeterminate. We detail some methods
to overcome this problem in the following sections.

Our choice to keep the bait to prey association is also subjective,
and we note that different investigators are likely to make different
choices as the nature of this interaction graph.

\subsection{Estimation of the Protein Complex Interactome}

The explicit details of the procedures and processes are provided in
the supplementary data. In brief, we assembled data from available
public databases (GOA and MIPS) that report information on known, or
predicted protein complexes and assembled that with protein complexes
predicted from experimental data \citep{Ho, Gavin, Krogan} using the
methodology of \cite{Scholtens, ScholtensVidalGent}. We chose to
process all experimental data uniformly, and attempted to exclude
estimates from GOA and MIPS that were based on the high-throughput
experimental co-precipitation experiments.

%%FIXME: Tony, here we need the number of complex estimates from GO, MIPS
%%and each of the three experiments, all separate in one table. You
%%should get a more recent (the most recent) GO build from Ting as I
%%think that there are some substantial improvements.

\subsubsection{Merging of the Data-Sets}

After the protein complex membership data-sets have been obtained and
transformed into an incidence matrix representation, the next step is
the construction of an aggregate protein complex interactome by
combining the separate incidence matrices into one single
non-redundant bipartite graph incidence matrix. The process of merging
the smaller incidence matrices requires that we compare each predicted
protein complex with all other protein complexes. Before proceeding,
we introduce a small lemma that simplifies the merging process.

\begin{Lem}
Let $C_1,\ldots,C_n$ be non-trivial finite sets which need not be
disjoint. Comparing all the sets $(C_1,\ldots,C_n)$ and removing all 
sets $C_{l_m}$ properly contained in any $C_i$ $\forall i \in [1,n]$
is an associative process, so it can be reduced to pairwise comparisons 
in any order.
\end{Lem}

\begin{proof}
  The proof of the lemma is a clear application to the fact that set 
  inclusion is a transitive property. For instance, suppose we have
  three sets $\{C_i, C_j, C_k\}$. We may assume that $C_j \subseteqq C_i$ 
  for otherwise, we do not delete $C_j$. If we first compare $C_j$ 
  with $C_i$, we would immediately delete $C_j$ when we see the proper
  inclusion. Now we can also suppose that $C_k \subseteqq C_j$. Then 
  we must have $C_k \subseteqq C_i$, and so it still be removed. 
  We have shown comparisons to be independent of ordering and therefore,
  is an associative process.
\end{proof}

With the an application to the lemma, we can merge the graph incidence
matrices iteratively by combining them pairwise. As an example, we
detail how the protein complexes of GO, $I_{g}$, and of MIPS, $I_{m}$,
are merged. We define the merging of two interactomes as not only
combining the protein complexes from $I_{g}$ to $I_{m}$, but also
removing protein complexes which are common to both interactomes. We
note that we can also remove a protein complex of $I_{g}$ which is
sub-complex to a complex of $I_{m}$ (or vice versa), and the process
remains independent of ordering. Once the matrices $I_g$ and $I_m$ are
merged into $I_{gm}$, say, we can repeat the process described above
by merging $I_{gm}$ with another incidence matrix. This process is
repeated until all estimates are merged. We will refer to the
resulting aggregate interactome as $I_c$.


\subsection{Binary Interaction Interactome}

%%FIXME: Tony are you sure that the Miller data are not in the newest
%%IntAct data?

%%It seems that we have not made use of the predicted interaction data
%% - should we?

To estimate a binary interaction interactome, $I_b$, we made use of
available data, largely based on the Y2H \citep{Y2H}. We considered
two distinct sources of data, one being experimental data and the
other being binary interaction predictions made by
\cite{binaryppintpred}. We used two sources for experimental data: the
\texttt{IntAct} repository and the experimental data obtained from
(cite Miller et al). We choose to use the IntAct database as the
primary source for Y2H data since it is quite complete, well
documented and provides the data in a form that is relatively easy to
process. The traditional Y2H methods fail to detect interactions
between membrane bound proteins; (cite Miller et al) developed a
variant of the Y2H methodology to overcomes these difficulties.

One of the difficulties in interpreting Y2H data arises from the
interpretation of missing edges, or equivalently missing
interactions. These can arise for a number of different reasons, false
positive and false negative results being among them. But a
potentially larger source of misinformation comes from a
misinterpretation of the meaning of the lack of an edge in the
observed data. If the bait protein is exposed to a complete set of
prey (i.e. all proteins in the organism are modified with the addition
of an activation domain) then the lack of an edge between the bait and
a true interactor is a false negative observation. But if that edge
was not tested, because the other protein did not have an activation
domain then the lack of an observed edge does not represent a false
negative result, but rather it is a missing value. 

For these reasons and to make the data more comparable we
restricted our analysis to the XXX data sets reported in IntAct that
appeared to be genome wide (i.e. at least 80\% of the genome was
tested as prey). We set a lower bound for the number of baits used at
the threshold of twenty proteins and note that due to the manner in
which the data are reported we will not know about baits for which no
interactions were detected. 

% The situation becomes problematic when merging the membrane bound protein
% interaction data with that of the \texttt{IntAct} protein interaction
% data. The genome-wide prey condition restricts the number of
% experimental data-sets we use from the \texttt{IntAct} repository,
% and also eliminates the data-set obtained from (cite
%Miller).

%%What should we do about this???


\section{The estimates}

We have now created two fundamentally different \textit{in silico}
interactomes, $I_c$ and $I_b$ by using the resources obtained from the
R package \Rpackage{ScISI}. From these two estimates, we can generate
a list of general summary statistics that offer a more qualitative
view of these interactomes.

\subsection{Summary Statistics for the Protein Complex Interactome}

<<ScISIdata, echo = FALSE, results = hide>>=
data(ScISI)
@ 

A good starting point in dissecting $I_c$ looks at how many genes are 
expressed from S. cerevisiae genome. In $I_c$, we can see that there is
\Sexpr{nrow(ScISI)} uniquely expressed genes which translates to 
\Sexpr{nrow(ScISI)} uniquely functional proteins participating within 
some non-trivial protein complex. In addition to the expressed genes,
we see that there is \Sexpr{ncol(ScISI)} distinct protein complexes
within the cell. 

\begin{figure}[p]
\centering
<<distPlot, fig=TRUE, echo=FALSE>>=
par(mfrow = c(1,2))
cS <- table(colSums(ScISI))
size <- as.numeric(names(cS))
repet <- as.vector(cS)
freq = rep(size, repet)
hist(freq, xlab = "ScISI Protein Complex Cardinality", ylab = "Frequency",
     main = "Complex Cardinality")

rS <- table(rowSums(ScISI))
size <- as.numeric(names(rS))
repet <- as.vector(rS)
freq = rep(size, repet)
hist(freq, xlab = "Complex Membership per Protein", ylab = "Frequency",
     main = "Complex Membership")
par(mfrow = c(1,1))
@
\label{distHist}
\caption{Distribution of Complex Cardinality and Distribution of Complex 
Membership per Protein}
\end{figure}

Not limited to the number of expressed genes and protein complexes, we
can investigate both the distribution of the size of the protein
complexes as well as the distribution of the number of complexes to
which each protein belongs. Figure~\ref{distHist} shows gives us these
distributions. We can see from each histogram that the vast majority
of proteins participate in relatively few protein complexes and also
that an overwhelming number of protein complexes contain relatively
few proteins.

<<testStuff, echo=FALSE, results=hide>>=
data("redundantM")
data("subCompM")
@ 

We can also inspect each pair of protein complexes from each of the
data repositories. In table~\ref{ta:Repetitions}, we have the number
of protein complexes which are common between each pair of
databases. For instance, the number of common protein complexes
between the GO and MIPS repositories is \Sexpr{redundantM[1,2]}, and
this is a relatively small overlap between protein complexes which
have been verified. It comes as no surprise that the overlap of
putative protein complexes estimated from the high through-put data
via \Rpackage{apComplex} with each other as well as with GO and MIPS
is particularly small. While this fact does not repudiate these
putative protein complexes, it does signal the need for further
validation.

<<redundantSubC, eval = TRUE, echo=FALSE, results=tex >>=
xtable(redundantM, display = c("s","d","d","d","d","d"), 
    label = "ta:Repetitions", 
    caption="Number of repetitive Protein Complexes") 

xtable(subCompM, display = c("s","d","d","d","d","d"),
       label = "ta:subComplexes", 
caption = "Number of Protein Sub-Complexes: the rows indicates the data repository
where the sub-complexes belong and the columns the over-all complex.")
@ 

The table~\ref{ta:subComplexes} accounts for the number of protein 
complexes from one database which are proper sub-complexes of protein
complexes in the other databases. This table is much less sparse than
table~\ref{ta:Repetitions}. These cross-references show one limitation
discussed by \citep{Scholtens}; the fact that \Rpackage{apComplex} 
is highly sensative the false negative observations in the AP-MS 
technology. From table~\ref{ta:subComplexes} and this limitation, we 
can reasonably assume that those putative estimates by
\Rpackage{apComplex} which are proper sub-complexes to verified 
complexes are ideal candidates for further computational work. 

\subsection{Summary Statistics for the Protein Binary Interactome}

<<getY2H, eval=TRUE, echo=FALSE, results=hide>>=
library("y2hStat")
data(y2hSysGW)
data(y2h)
data(y2hGW)
@ 

Much like the analysis for $I_c$, we begin the dissection of $I_b$ by
investigating the number of baits sampled. From the restrictions
described above, we made use of only \Sexpr{length(y2hSysGW)} of the
experimental data-sets, and from these, a total of
\Sexpr{length(unique(unlist(lapply(y2hSysGW, function(x) names(x)))))}
unique proteins were tagged with binding domains and became function
bait proteins. While the prey population is genome wide, only
\Sexpr{length(unique(unlist(y2hSysGW)))} unique proteins interacted
with some particular bait protein(s) and became function prey
proteins.

The identifiability problem is plays an important role in $I_b$. A
small calculation shows that that mapping from the IntAct acension
codes could identify
\Sexpr{length(grep("Y",unique(unlist(lapply(y2hSysGW, function(x) names(x))))))} bait proteins and \Sexpr{length(grep("Y", (unique(unlist(y2hSysGW)))))} 
prey proteins by their respective 
SGNs. These un-defined mappings percolate through $I_b$, but we have
left them in the interactome because their inclusion has no effect on
our computational analysis (we remark that removal is trivial and the
choice is left to the user).

<<someY2HStats, eval=TRUE, echo=FALSE, results=hide>>=

baitsUsed <- unique(unlist(sapply(y2hGW, function(x) {sapply(x,
                                                             function(y)
                                                             y[1])}))) 
translation <- y2h$baitsSystematic[baitsUsed]
numMulNameBaits <- sapply(translation, length)

preysUsed <- unique(unlist(y2hGW))
translation2 <- y2h$preysSystematic[preysUsed]
numMulNamePreys <- sapply(translation2, length)
@ 

\begin{figure}[p]
\centering
<<multiPlot, fig=TRUE, echo=FALSE>>=
par(mfrow = c(1,2))
multi <- table(numMulNameBaits)
size <- as.numeric(names(multi))
repet <- as.vector(multi)
freq <- rep(size, repet)
hist(freq, xlab = "Number of SGN", ylab = "Frequency",
     main = "Bait Mapping Multiplicity")


multi <- table(numMulNamePreys)
size <- as.numeric(names(multi))
repet <- as.vector(multi)
freq <- rep(size, repet)
hist(freq, xlab = "Number of SGN", ylab = "Frequency",
     main = "Prey Mapping Multiplicity")

par(mfrow = c(1,1))
@
\label{multiHist}
\caption{Histograms giving the multiplicity of the bait and 
prey protein mappings from the IntAct Codes to SGNs.}
\end{figure}

%%%What else shoule be said? I don't like the histograms because 
%%it does not give any indication if the mapping is has >2 outcomes

   
\section{Validation}

There are a number of different methods that can be used to validate
the ScISI prediction. Our basic premise is that the complex is a
collection of physically connected proteins. While, in some cases this
may not be strictly true and there could be two or more subcomponents
that do not physically interact we will demonstrate that such cases
have little impact on the validation methods we employ and hence are,
to a large extent, covered by our proposals.

If all physical interactions between proteins were known, then the
problem would be somewhat straightforward to address. One would simply
take a predicted complex and determine whether the constituent
elements interact and deem the complex \textit{viable} if it could be
configured as a single connected component. However, very few
interactions are known, and it is this incompleteness that requires us
to make inference about whether the prediction is likely to be
valid. Unfortunately the topology of the complexes (the set of actual
true interactions) is unknown and hence cannot itself be used to
assess the observed data. We do not know how many edges (physical
interactions) there should be, and whether the complex follows a spoke
topology, a matrix topology or any of the many other
possibilities. 

We also note that one of the alternatives that some have used, which
is to compare their predictions to the published, and to some extent
verified complexes described in GO and MIPS is not available to us. We
used those sources in construction of our estimate and hence cannot
validate against them. Instead we make use of two other sources of
data. One being the set of observed yeast two-hybrid (Y2H) experiments
available from IntAct and the other being a set of predicted
protein-protein interactions from \cite{PPIpred}.

Before describing the data we briefly discuss the statistical issues
involved. In most cases only a subset of the genes were assayed, or
considered and it will be important to restrict the computations to 

A working hypothesis is that every protein complex is a single
connected component, so that when the complex is functional it is a
single unit. This does not mean that every protein physically
interacts with every other protein, in some cases a protein may
interact with only one other protein, while in other cases it may
interact with many. 

\subsection{Y2H Validation}

%%FIXME: Tony, please report here 1) number of experiments, that we
%%are dropping those with less than 20 interactions since there is
%%likely to be substantial experiment to experiment variability.
%% Maybe a table with experiments, system (Gal4), number of baits
%% and whether it is genome wide.

We rely on Y2H data provided by IntAct (citation) and compute a number
of summary statistics for the complex estimates in the ScISI. We
obtained data on eight Y2H experiments comprising ZZZ unique reported bait 
proteins and ZZZ unique reported prey proteins. It is important to 
recognize the fact that information can
only be obtained for complex estimates that contain one or more Y2H
bait proteins. And since the actually physical interactions are not
known, the topology of the Y2H graph is also not known.
%%FIXME: Tony describe the number of experiments here, a table
%%describing something about baits into

Any bait protein in a Y2H experiment should find one or more complex
co-members provided those co-members were available as prey in the Y2H
experiment. This proviso indicates one of the true weaknesses of Y2H
data as they are often reported since only information about the prey
that were detected and not the prey that were tested is
reported. Thus, the absence of an edge is not as informative as it
could be. We do not know if it was tested and not found or not tested,
and these are very different things.


\subsection{PPI Predictions}

<<ppipred, echo=FALSE, results=hide>>=
library("y2hStat")
data(ppipred)
@ 

\cite{PPIpred} report \Sexpr{nrow(ppipred)} pairwise predictions
between proteins in \textit{S. cerevisiae}, with a presumed 
false positive rate of $3E-4$ and a presumed false negative rate of
$0.85$. The predictions are accompanied by a probability and one can
select only those predicted interactions according to this
probability. We have chosen to use only those interactions where the
probability is at least $0.5$. This greatly reduces the number of 
predicted interactions and also greatly reduces the number of
proteins that were observed. 

When using these data, only proteins that are predicted to be involved
in at least one interaction are \textit{observable}. This changes as
the probability cut-off changes and can be interpreted as the
sensitivity of the analysis. This phenomenon is no different than that
observed with different experimental procedures, not all interactions can
be detected with all procedures and one must make some efforts to determine
which interactions can be detected.

Once the observable proteins have been determined they can be use to
compute the different summary statistics discussed in
Section~\ref{sec:summary}. 

For each protein complex, $C$, we divide its constituent proteins into two
sets, those that are observable, denoted $U$ and those that are not,
$\overline{U} = C \\ U$. If $P \in U$, then we expect there to be 
an edge between $P$ and some other member of $C$, but we will only
observe this edge if it is to another member of $U$. 


\subsection{Summary Statistics}
\label{sec:summary}

As stated above, our definition of a multiprotein complex presumes
that every member is physically linked to at least one other member of
the complex. Stated in graph theoretic terms we assume that the graph
of a complex where nodes represent proteins and edges represent
physical interactions is connected. 

To develop appropriate statistical methods we describe the same
setting in a slightly more abstract notation. For a given complex,
$C$, say, consisting of $n$ proteins the graph that represents this
complex must be connected, by our definition of a complex. Hence,
there are between $n-1$ and $n(n-1)/2$ edges. We can conceptualize
this in the following way. Consider an urn that contains balls, each
ball represents on possible edge, or binary interaction. For $C$ there
are $n(n-1)/2$ possible edges and hence there are that many balls in the
urn. Some of the balls represent edges that truly exist in $C$, and these
are colored black, the remainder of the balls are white.

We will, in some cases, use the notation $G_C = (V_C, E_C)$, to denote
the graph induced by $C$. We note that in this graph the edges are not
directed and there are no self-loops. That is, we assume that no
protein in $C$ has an edge to itself, in part this makes the math
simpler and in part it reflects the technology. While AP-MS can
determine the constituent elements of complex it cannot directly
ascertain their multiplicity.

The first problem we address is estimating the number of black balls
in the urn. We label this unknown quantity $X$. The basis for this
estimation is the sampling of some nodes of $C$ and determining which
other members of $C$ they are observed to be connected to. We will
presume that the nodes that are sampled are a simple random sample
from the population $V_C$, but note that this is not always the case
in real experiments. Let $k$ denote the number of nodes sampled and
let $n= |V_C|$, when needed we will also use $n_C$. Further, let $x$
denote the number of distinct edges found based on the sampling of $k$
nodes. 

To determine how many edges were tested we note that the first node is
compared to the remaining $n-1$, the second to the remaining $n-2$ and
so on. So, the number of edges tested is $[(n-1)+(n-2)+\ldots+(n-k)]$.
If we sample all nodes then the sequence is, 
\[
\sum_{j=1}^{n-1} (n-j) = \sum_{j=1}^{n-1} j = \frac{(n-1)n}{2},
\]
where we have made use of the well known relationship regarding the sum
of the first $n$ integers. And hence, sampling all nodes does in fact
result in the inspection of all edges.

A widely used estimate of $X$ arises from equating the observed
proportion of edges in the sample, with the unknown proportion in the
population. That is,
\begin{eqnarray*}
\frac{\hat{X}}{n(n-1)/2} & = & \frac{x}{[(n-1)+(n-2)+\ldots+(n-k)]} \\
\hat{X} & = & \frac{ x (n)(n-1)}{2 [(n-1)+(n-2)+\ldots+(n-k)]} \\
\end{eqnarray*}

A second estimator can be derived from the following argument. If the
nodes that are sampled represent a random sample from the population
of nodes, $V_C$, then the observed mean degree is an unbiased estimate
of the population mean degree. The population mean degree times $n$
divided by 2 is an estimate of the number of edges. For each sampled
node we let $d_i$ denote its observed degree. So, we have
\[
\tilde{X} = \frac{ n \sum_{i=1}^k d_i}{2k}.
\]

The two estimators are quite similar, and sometimes identical.

At this point we have merely estimated the number of edges in $G_C$,
from that we make the next step of assessing whether or not $G_C$ is
connected. There are a number of factors that influence that
determination and we provide simulation examples and evidence of how
these different factors can be used.

First, the larger $X$ the more likely the graph is connected. Next, we
can examine the number of unique nodes that were selected either as
part of the $k$ nodes used to probe the graph or as the other ends of
the detected edges. The more nodes detected this way, the higher the
probability that the graph is connected. And finally, we can take the
observed subgraph, induced by our sampling, and determine how many
more edges are needed to obtain a connected graph; the fewer the more
likely it is that the true graph is connected.

We also explore a simulation approach to assessing whether the
underlying graph is connected. Given the observed edges for the $k$
query nodes and the estimated number of edges in the graph, $X$, one
can simulate the remainder of the graph by drawing from the remaining
edges. For each sample, one can assess whether the graph is
connected. Repeating this a large number of times yields some number
of connected graphs and some number of unconnected graphs. The
relative proportions can be used to assign a probability that the
underlying graph is connected. This approach can also be loosened if
desired and other measures made on the simulated graphs.

% If the graph is connected there must be at least $n-1$ black balls,
% but simply having $n-1$ black balls will not ensure that the graph is
% connected. However, the more black balls there are, the more likely it
% is that the graph is connected. We use this observation as the basis
% for our proposals. We let $X$ represent the true number of black balls
% in the urn and note that this number is unknown and hence needs to be
% estimated.
%%FIXME: note that if we associate a pair of names with each ball then
%%the graph is connected when the union of those names is of size n,
%%but it is not obvious to me how we might use that.

% Both sampling schemes, Y2H and PPI predictions, can be viewed as
% drawing some number of balls from the urn. For each ball drawn we do
% know if that ball represents a true edge (and hence is black) or not,
% provided we are willing to presume that there are no false positives
% and no false negatives. Thus the observed data can be interpreted as
% having $k$ black balls among $l$ draws. A simple estimate of $X$ can
% then be found by equating the proportion of black balls in the sample
% with the proportion of black balls in the population. And hence that, 
% \[
% \hat{X} = \frac{k n (n-1)}{l},
% \]
% and one can also easily show that $E(k n(n-1) / l) = X$, under a
% random sampling scheme and that
% $var(k n(n-1)/l ) = X(n(n-1) - X) (n(n-1) - l) / [l (n(n-1) - 1)]$,
% although one suspects that better estimates can be obtained.

% A potentially better idea is to use the fact that for some types of sampling,
% especially that of Y2H, the observed out-degree is an unbiased
% estimate of the true outdegree. That, times the number of nodes/genes
% in the complex gives an estimate of the number of edges, which has
% the advantage of being unbiased. One suspects that better estimates
% of the number of edges could be achieved by using in-degree as well,
% but those estimates are biased.

Once we address real experimental data, the situation changes and
becomes more problematic. Some alternative approaches will be needed. 
In addition we propose four different summary statistics and
investigate their behavior using the ScISI and the available Y2H data.

\begin{enumerate}
\item For a given protein complex, $C_i$, find all complex members,
  $P_j$ that were used as a bait in some Y2H experiment. For these,
  compute the proportion that found at least one other member of the
  complex. If $P_j$ was used as a bait in more than one Y2H experiment
  do not double count, but take any positive result as positive.
\item For a given protein complex, $C_i$, find the average out-degree
  of all bait proteins, again avoid double counting, in this case by
  taking the maximum out-degree. Divide this by the complex size.
\item Given the number of proteins in a complex that are detected as
  either bait or prey, find the number that are connected to at least
  one other complex co-member. [Should this be the proportion of
  complex members that are connected to at least one other complex
  member?] 
\item Compute the ratio of the number of edges needed to make the
  complex connected, by adding to the observed edges, divided by the
  minimum number of edges needed to create a connected graph from the
  complex. This is essentially a measure of incompleteness.
\end{enumerate}


\section{Discussion}

What did we learn,

\bibliographystyle{plainnat}
\bibliography{bioc}

\end{document}
