%
% NOTE -- ONLY EDIT ScISI.Rnw!!!
%
%\VignetteIndexEntry{ScISI Working Paper}
%\VignetteDepends{}
%\VignetteKeywords{Interactome}
%\VignettePackage{ScISI}
\documentclass{article}

\usepackage{hyperref}

\textwidth=6.2in
\textheight=8.5in
%\parskip=.3cm
\oddsidemargin=.1in
\evensidemargin=.1in
\headheight=-.3in

\newcommand{\Rfunction}[1]{{\texttt{#1}}}
\newcommand{\Robject}[1]{{\texttt{#1}}}
\newcommand{\Rpackage}[1]{{\textsf{#1}}}
\newcommand{\Rmethod}[1]{{\texttt{#1}}}
\newcommand{\Rfunarg}[1]{{\texttt{#1}}}
\newcommand{\Rclass}[1]{{\textit{#1}}}



\newcommand{\classdef}[1]{%
  {\em #1}
}

\begin{document}
\title{Creating in Silico Interactomes}
\maketitle

\section{Introduction}

Protein interactions form a fundamental part of biology and are
relevant to our understanding of complex biological processes and to
deciphering the different roles played by genes and proteins. A number
of high throughput experiments have been carried out with the goal of
helping to elucidate the nature of the interactions and the number
of interactions.
present \cite{Gavin, Ho, Krogan, Ito, Uetz}. Yet, for all organisms or
cells our knowledge is very incomplete. Hence, not only do we need
much more experiments but we also need funcdamentally different 
experiments to fully reveal the different network structures. However, 
such experiments are expensive and time consuming to carry out; hence,
the need for simulation models that can help to identify
the viability of experimental procedures, to aid in integrating data
from disparate sources, and to provide a sound basis for the
development of new methodology to help design experiments and to
analyze the resultant data. In this paper we outline one approach to
the creation of \textit{in silica interactomes}. (in silica???)

We begin by distinguishing between many different types of
interactomes with a few general comments and make many of the
terms and ideas more explicit in later sections of this report. There
are interactomes that model direct binary interactions and hence are
related to experiments using the yeast two hybrid (Y2H) system
\cite{FIXME:citatins here}. A second type of interactome is one that
models protein complex co-membership. In this model, two proteins are
said to interact if they are consitutent members of some protein complex 
under the conditions of interest. This second type of interactome is
more closely related to co-precipitation or affinity-purificatin
experiments \cite{FIXME:somecitations}. We note that proteins which
interact in the second sense, need not interact in the first sense;
hence, we separate these two different, but related, concepts.

In this paper we consider the problem of estimating an \textit{in
  silico} interactome for \textit{Saccharomyces cerevisiae} and we
describe five different roles where such a computational device can be
used.  First, the \textit{in silico} interactome provides a tool
that can be used to mimic wet-lab experiments; these \textit{in
  silico} experiments can help to confirm the viability of the wet-lab
experimental procedures by allowing investigators to alter
experimental conditions (both stochastic and systematic errors can be
modified) and to measure how the outputs change as the error rates
do. Second, the \textit{in silico} interactome provides a tool that
can generate multiple data sets, under different conditions, so that
the statistical properties of different of proposed estimation
procedures can be directly compared. A third use for the \textit{in
  silico} interactome is to use it to help develop tools and
strategies for small scale experiments that will probe the interactome
at a very detailed level. For example, a reasonable strategy to
investigate a coherent set of protein complexes is to first select a
small number of candidates and explore their interactions, say via
affinity purification (co-precipitation) experiments. Once the first
experiment has been collected one would like to analyze the data and
determine an optimal, or nearly optimal set of baits for future
experiments. A fourth use of an \textit{in silico}
interactome is to study the effect of perturbations in a network.
Such perturbations could be caused by evolution (and hence could become
part of the study of evolutionary pressure on the interactome) or due 
to drug treatments (and hence would become part of a
clinical investigation into the likely effects of drug treatments). The
fifth use for such a tool is to explore and understand the effects of
different sampling paradigms on the inferences that can drawn from
different wet-lab experiments. We note that there are a number of
recent papers (FIXME:citations) that raise many important issues in
this regard.

While the estimated interactome is itself of some great interest, we
note that it is more important to describe the procedure by which the
\textit{in silico interactome} is produced. This methodology is more
important than the estimate, since from it new estimates can easily be
produced, and thus it is this construction that is our focus. There are many
reasons why the answer is of less importance, among them the fact that
the methodology can be applied to many organisms and tissue types. We
also note that genomic science is rapidly evolving and that new data
sources will continuously come on-line. Hence there must be a
mechanism for updating the estimated interactome. There are many
different and disparate data sources and the system should be
sufficiently flexible to allow for the addition of data from new and
potentially different experimental techniques. Finally we note that
different investigators will want to modify the interactome, either
with localy produced, though not yet public, data, or to satisfy personal
beliefs. We have developed a paradigm that allows investigators to
create their own personalized estimates, to share interactomes and to
build, revise and update interactomes.

We are aware of the very dynamic nature of protein interactions and
that different interactions occur at different times and under
different conditions. In this paper we consider the description of an
interactome that can be thought of as representing the state of a
single type cell (our example will be based on
\textit{S. cerevisiae} where there is a single cell, but where
different conditions could conceivably result in different
interactomes). We leave the problem of representing state-dependent
interactomes for a separate report.

% We have carried out a number of different investigations of these data
% \cite{Scholtens, ScholtensVidalGent} which have focused on the
% estimation of protein complex co-memberships. 

In this paper we first define the notion of a biological interactome and 
give various examples of such interactomes. We then present a  
computational tool (\textit{ScISI}) 
we have built to search for and to collect protein complex membership 
data from various protein interaction repositories. Next we describe 
the primary data sources used by \textit{ScISI} in constructing the 
\textit{in silico interactome}. We then describe how those data
sources are combined to provide a comprehensive interactome that can
be used as an input for the various computational tasks described
above. Finally we report on some \textit{in silico} experiments that
can be performed using the estimated interactome.

In our discussion we do not distinguish between the term gene and the
term protein, since the available resolution is not fine enough and
equating these concepts leads to a more comprehensible dialog. The
methodology is sufficiently general to be able to address the more
complicated situation of different protein/polypeptide variants should
they arise.



\section{Biological Interactomes}

In the broadest sense, an interactome is a set of elements $P$ where 
the individual elements of $P$ have some associtation with one another.
Consider the example where $P$ is the United States population and 
association is defined as immediate family members, i.e. either parent
to child or sibling to sibling. In this example, a person is neither 
associated with his/her grandparent nor associated with his/her
aunts/uncles. Now let us consider a different interactome where 
$P$ remains the same set but association is defined as familial relations.
With this definition, a person is now associated to his/her grandparent, etc.
It is trivial to see that association in the second example implies 
association in the first, but the 
converse is not true. These two examples clearly illustrates the need 
for the association within a particular interactome to be well defined.

We can relate the two previous population interactomes to biological
interactomes. Now let $P$ be the set of genes within the genome of some cell
or tissue. Let $Y$ be the interactome on the set $P$ where the association
is direct binary interactions between genes, i.e. two genes $g_1$ and $g_2$
are said to be associated if and only if the induced protein of $g_1$ and the 
induced protein of $g_2$ directly interact for some bio-physical or bio-chemical 
process. We remark that a gene may be associated with itself if it induces a 
homo-dimer. Next, let $A$ be the interactome on the set $P$ where the association is
protein complex co-membership, i.e. two genes $g_3$ and $g_4$ are associated
if and only if their respective induced proteins form constituent parts to
some protein complex that has some biological objective. 

These two biological interactomes are related. It is clear from 
interactome $Y$'s definition of association that any two associated
genes within $Y$ will also be associated within the interactome $A$. 
The reverse relationship,
however, is not valid. Not all co-members of a particular protein 
complex need to interact with all other members. These two examples 
highlight the subtle differences between two distinct interactomes 
and underscores the necessity to keeping distinct interactomes separate. 

$Y$ and $A$ describe two instances of protein to protein interactomes,
but there are a number of other such interactomes with biological 
relevance. Let $S$ denote the interactome on $P$ where association 
between two genes is that of synthetic lethality. Let $E$ be the 
interactome on the set $P$ where association between two genes is 
defined as co-expression. While all four interactomes are 
inter-related, no two are equivalent, and thus, all conclusions must 
be drawn independently for each interactome. 

In this paper, we will build an \textit{in silico interactome}
for Saccharomyces cervisiae (denoted $I$) for a fixed set
of environmental conditions. The elements of $I$
are the disjoint union of two sets: the first is the set $P$ of genes
within the genome of a yeast cell; the second is the set $C$ of all protein
complexes of yeast for a specific cell type. Association is defined as
protein complex membership, i.e. gene $g$ is associated with protein
complex $c$ if and only if the induced protein of $g$ is a constituent
member of $c$. This definition implies that elements of $P$ are only 
associated with elements of $C$ and conversely, but no element can 
be associated with other elements of the same type. 

\section{Materials and Methods}

In this section we first describe what we mean by a protein complex
and describe different models for representing these data. Next we give
a consise description of the computational tool \textit{ScISI} and show
its dynamic properties in constructing different complex membership
interactomes. We then
describe the process of creating an \textit{in silico} interactome
via \textit{ScISI} based on two public meta-data sources and three 
experimental data sets 
by combining purification of protein complexes with identification of 
their individual componenets via mass spectroscopy.


\subsection{Protein complexes and their representations}

%%FIXME: a few pictures would surely help readers
A protein complex (defined for this paper) is two or more
proteins that physically interact for the purposes of carrying out
some biological objective. There is no need for all members of a
complex to interact with each one another, but all the proteins do need
to combine to form a macro-proteomic structure. We further note that 
some proteins will be involved in many complexes
while others will be involved in relatively few, and one corollary 
to this remark is that the relative size of the protein complexes 
range from two proteins to more than fifty. 

This definition of a protein complex is motivated by the various 
technologies that assay levels of protein complex composition such 
as Affinity Purification - Mass Spectroscopy and 
Co-immunoprecipitation. It is important to note that these 
technologies only assay protein complex composition of cells
given a pre-described environment. Working within the confines of these 
technologies, our definition for protein complexes becomes 
increasingly strict. There are structures within the cytoskeleton,
for instance, which fit within a more generic setting of a complex; 
these structures are never pulled down by the mentioned 
technologies, so we have dis-allowed their membership as a protein
complex. 

Because our definition for protein complexes is experimentally 
driven, we need to define a convention with respect to protein
sub-complexes. A protein sub-complex is itself a  
complex that is entirely contained within another protein 
complex. We remark that a protein sub-complex under certain 
cell conditions could be a proper complex in other
cell conditions. We define the convention that all  
sub-complexes will be masked by their respective proper 
protein complexes, and therefore, are not recorded. Thus our 
definition of protein complex is implicitly directed by the 
biological state of some particular cell or tissue, i.e. a 
complex can only be determined along with the context of the 
cell type.

%What should be said about sub-complexes? When sub-complexes exist
%there can be identifiability problems that arise due to the technology
%being used. FIXME: it would be good to give some sort of concrete
%example of what might go wrong. (Also, some protein complexes might
%be a sub-complex given other conditions...where should we draw the 
%line between when a complex is actually a sub-complex or not??? TC) 

The collection of proteins and protein complexes with affiliation data can
be represented as a large membership network or graph. We have chosen 
to represent the protein complex membership graph as
a bipartite graph, $B$. The proteins themselves index one set 
of nodes which we denote as $P$ while the protein complexes index
the other set of nodes which we denote as $C$. A protein specific
node $p_i$ is incident to a protein complex specific node $c_j$ if $p_i$ 
has membership in $c_j$. We remark that there is an equivalence between the categories
of bi-partite graphs and the categories of hyper-graphs (cite or supplementary).

To each bi-partite graph comes an associated incidence matrix representation. 
This incidence matrix representation, $M_B$, is a $\{0,1\}$-matrix where proteins
index the rows and protein complexes index the columns. The entries of $M_B$ 
correspond to the presence or absence of membership edges, for instance, the $(i,j)$ 
entry for $M_B$ is unity if protein $p_i$ is a constituent member of complex
$c_j$; the entry is zero otherwise. 

Because most computational methodologies desire this incidence matrix representation
rather than the actual bi-partite graph itself, the construction of the \textit{in silico}
interactome will be the construction of a particular incidence matrix.  

\subsection{ScISI}

With publicly released data available in a number of disparate sources, there is a 
need for computational tools to parse through the data-sets and collate them into 
one single, non-redundant, and meaningful database resource. To this end, we have 
built the $R$ package \Rpackage{ScISI} for the specific purpose of parsing through 
protein interaction resources and selecting protein complex information, i.e. 
the various documented protein complexes as well as their consistuent protein members.

\Rpackage{ScISI} takes a protein interaction repository and systematically parses
through that repository's database in search for key terms. Depending on the catalogue
of each repository, \Rpackage{ScISI} parses the data and searches (either by grep or
agrep) for three key terms: 1. the exact word ``complex''; 2. the suffix ``-ase''; 
and 3. the suffix ``-some''. We remark that the user can search for any term desired
though the three terms we have used are the defaults. 

Once the various databases have been parsed and the desired data sets stored, 
\Rpackage{ScISI} converts the overall collection of protein complexes into a
bi-partite graph incidence matrix where the rows are indexed by the standard
gene names and the columns are indexed by the protein complex identification 
codes (unique to each different repository). Therefore, \Rpackage{ScISI} has 
produced an \textit{in silico sub-interactome} (or a sub-bi-partite graph
of the bi-partite interactome graph).

While the overwhelming majority of protein interaction repositories store protein
complex data as some form of a catalogue (unique to the repository), there are 
a small number of repositories which store the bi-partite graph incidence graph 
matrix for the protein complex data. For these repositories, \Rpackage{ScISI} 
simply downloads these matrices.

Once the various protein complex membership data has been stored, and the various
bi-partite graph incidence matrices have been constructed, \Rpackage{ScISI} compares
all each of these matrices with one another and eliminates redundancy by removing 
multiple occurances of a particular protein complex; and based on our model and 
definition for protein complexes, \Rpackage{ScISI} also removes any complex if it
is contained entirely within another complex. The details of this merging process 
will be given in a later section.

Having streamlined a definition for protein complex and its computational representation
as well as given a description of the computational tool used, 
we move forward with the task of diciphering an \textit{in silico} interactome for S. 
cerevisae. To this end, we have gathered information from  
two online data sources, Gene Ontology and the Munich Information Center for Protein 
Sequences, as well as high-level protein complex estimates from three publicly available 
wet-lab experiments. 

\subsection{Online Repositories}

Using the R package \Rpackage{ScISI}, we have parsed through two online protein 
interaction repositories: Gene Ontology (GO) and the Munich Information Center for
Protein Sequences (MIPS). While the organization of these two online databases 
are quite different, \Rpackage{ScISI} identifies each repository and (using 
different functions and methods) from each, extracts the protein complexes.

As we have stated previously, \Rpackage{ScISI} parses through each repository
and searchs for the terms $complex$, $\verb1\\base\\B1$, and $verb1\\bsome\\B1$.
The first term is searched for either with $grep$ witin the MIPS database and with
$agrep$ within the GO database. The search functions within \Rpackage{ScISI} parse
the titles of each protein complex by $(a)grep$ and looks for the exact term
``complex'' (e.g. SAGA complex). The latter two terms are simply $grepped$ but 
contain perl regular expressions which serve as boundary conditions for this parse 
function within \Rpackage{ScISI}. These boundary conditions narrow the pool of 
return values (e.g. DNA-directed Polymerase II, holoenzyme or repairosome). 

A second boundary condition encoded within the parse functions of 
\Rpackage{ScISI} is the evidence codes used by both MIPS and GO (though
these codes are unique to each repository) to annotate individual 
proteins to the various protein complexes. \Rpackage{ScISI} can take a
list of evidence codes and dis-allow those proteins whose annotations
are based exclusively on this list. (Should we use actually use some
evidence codes when constructing the ISI???)

After parsing the GO repository (as of 17 December 2005), 208 protein 
complexes were extracted and stored. The MIPS repository (as of 17
December 2005) yields 141 protein complexes. The GO protein complexes
is built from 1,213 unique proteins while the MIPS protein complexes
contains 651 unique proteins. The protein complexes and its constituent
protein members are stored in a list type structure and must be further
processed.

Once a list of protein complexes is generated for a particular repository, 
\Rpackage{ScISI} transforms this list into a bi-partite graph incidence
matrix, i.e. an \textit{in silico sub-interactome}. These two matrices must
be examined both by themselves or cross referenced between one another to 
search for repeated protein complexes or for the prescence of protein 
sub-complexes. The details of this examination is left for a later section.

We remark that the complexes documented within Gene Ontology and the Munich
Information Center for Protein Sequences are annotated by many different 
experiments such as small scale protein complex verification experiments. Many
of these protein complexes have been exhaustively verified, and thus, 
these complexes have a significant level of accuracy with respect to their
constituent protein members. 

Some of the MIPS protein complexes have been uniquely identified by high 
through-put experimentation such as affinity purification and 
co-immunoprecipitation; we have, however, deliberately chosen not to 
extract these protein complexes. While high through-put data from AP-MS
specifically tries to ascertain protein complex composition, the data 
itself is overwhelmed by experimental error (both stochastic as well as
systematic). The protein complexes were estimated by clustering algorithms
decided by the individual investigator of the high through-put experiments,
and therefore, protein complexes identified via high through-put AP-MS
technologies lack uniformity and consistency. We have chosen one standard
algorithm to parse high through-put data and estimate protein complex
composition from the raw data.   

\subsection{AP-MS Experimental Data Repository}

In this section, we describe the experiments conducted by Gavin, et al; Ho, et al; and most recently, Krogan, et al. All three experiments are based on the high-throughput techonology of complex purification followed by identification by mass spectroscopy. Each affinity purification - mass spectronomic (AP-MS) experiment yielded a tremendous wealth of bait to prey associations, but we must take these data points with care because high through-put data is known to repot high levels of error in addition to true associations between proteins. 

In each of the three datasets, bait to prey associations were inferred by using various kinds of the same methodology: an affinity based process to localize groups of proteins (the associated prey proteins) that have high probability of sharing some complex co-membership with the bait protein followed by a purification procedure to eliminate non-associated proteins which might haven been pulled down due to some stochastic elements and a protein complex identification process relying on mass spectroscopy. Gavin, et al used the procedure of tandem affinity purification (TAP) where the TAP tag is generated by polymerase chain reaction (PCR). Ho, et al used a version of immunoprecipitation which they have called high-throughput mass sprctrometric protein complex identification (HMS-PCI). Krogan, et al used a C-terminal tandem affinity purification (TAP) tagging. Because these three experiments are based on bait to prey association for protein complex identification, the raw data can be systematically and uniformly analyzed by \textit{apComplex} for estimation of protein complex composition.

Analysis by \textit{apComplex} transforms the problem of biological protein complex estimation to a problem of estimating the presence of directed edges in a protein-protein complex co-membership (PPCCM) graph. Recall that two nodes are adjacent in the PPCCM graph if they are co-members for any protein complex. Under such an imposed structure, the identification of protein complexes is equivalent to the identification of maximal cliques in the PPCCM graph. %%Confirm it has been shown that the exhaustive search for maximal cliques within a graph is $np-hard$ based on the number of nodes within the network. Independent of the complexity in maximal clique mining, the PPCCM graph derived from the AP-MS technologies contains errors by embedding false positive (FP) and false negative (FN) bait to prey associations. These errors translate to the addition of FP directed edges and the deletion of FN directed edges respectively %%need picture.

With the wet-lab experimental data, \textit{apComplex} searches for Bait to Hit (BH) maximal complete subgraphs (since maximal complete subgraphs are defined to be maximal cliques) which all recipricated bait-bait edges as well as all unreciprocated bait-hit-only edges exist. When such BH maximal subgraphs are found, \textit{apComplex} uses a statistical paradigm to determine the presence or absence of directed edges (particularly edges from hit-only proteins back towards those bait proteins which have unrecipricated association)
. The details concerning the statistical model upon which \textit{apComplex} is based can be found in \cite{Scholtens, Gentleman}.

The estimates derived from \textit{apComplex} can be divided into three groups. The MBME protein complex estimates are those derived from BH maximal subgraphs where multiple bait proteins (represented as nodes) and multiple edges (representing hits) are present. The SBMH protein complex estimates describe those complexes with only one single bait present but multiple hits (edges). Lastly, the UnRBB estimates consist of only two nodes (both bait proteins) where only one unreciprocated directed edge exists between them. It is quite clear that the MBME complexes have the highest probability of estimating true protein complexes since they are based on more biological information. For the construction of the \textit{in silico} interactome, we have elected to use only the MBME estimates because of its relative likelihood for accuracy.

Data from the AP-MS experiment have been filtered to yield protein complexes and stored in databases such as MIPS; the process for each experiment, however, is unique to each experiment. %%FixMe Each filtering process is based upon some statistical clustering paradigm that takes the underlying biological conditions with varing degrees. Without a uniform methodology to estimate protein complex co-membership, any \textit{in silico} interactome built from the estimated complexes would be suspect. \textit{apComplex} integrates a significant portion of the biological processes into its statistical model in estimating protein complexes. Thus by using \textit{apComplex} as the unique estimation algorithm, we can not only reduce the variability in confidence but also increase the overall confidence for the estimates of the protein complexes.


\subsection{Merging the Data}

Having obtained the data from all the five disparate data souces, we 
must combine all of the data into one non-redundant aggregate 
compilation. Merging the multiple sub-interactomes into one is done 
iteratively by combining two sub-interactomes at once.

Let $M$ and $G$ be two interactomes that we wish to merge. We define
the merging of two interactomes as not only combining the protein 
complexes from $M$ to $G$, but also removing protein complexes 
which are common to both interactomes. In addition to removing 
redundant protein complexes, we have removed any protein complex 
of $M$ that is a sub-complex of $G$ and vice versa. 

Once the sub-interatomes $M$ and $G$ are merged into a comprehensive
aggregate interactome $A$, we can repeat the process described above
by merging $A$ with another sub-interactome $S$. 

After a finite number of steps in the iterative process, we have 
constructed the \textit{in silico interactome}, $I$. The 
characteristics to this interactome is that protein complexes 
are either present or absent (we do not attempt to estimate the
concentration of any protein complex) and protein sub-complexes 
are always un-identified. 

%%Talk about the number of redundancies and sub-comps

%%The stuff written below does not seem to fit in the methods section,
%%but more in a discussion section??? Seems like the intro to a dynamic
%%algorithm paper or something???

%When estimates of the interactome are available from different sources
%we must devise a method for combining them to provide a larger and
%more general estimate. How we do this depends a bit on the estimates
%themselves. We leave to another time a discussion of how to address
%issues of sytematic and stochastic errors and rather consider the
%problem of having two different estimates of an interactome that we
%want to combine into a non-redundant joint estimate. The two different
%estimates may come from different sources and hence could overlap by a
%little or a lot. 

%The so-called \textit{alignment} problem arises because some
%complexes might not be labeled or are  unambiguously identified. The genes (or
%proteins) that make them up, are well identified but the collections
%are not. So, one of the first steps is to identify redundancy and to
%eliminate it. For simplicity we presume that the complexes in one
%estimate are labeled $C_1, \ldots, C_k$ and in the other
%$K_1,\ldots,K_l$, where $k$ and $l$ are generally not the same.
%We will discuss set theoretic relationships between the $C_i$ and
%$K_j$ in terms of the proteins that constitute the complexes. Thus,
%the statement, $C_i = K_j$, means that all proteins in $C_i$ are in
%$K_j$ and the converse also holds. The statement, $C_i \subset K_j$
%indicates that $C_i$ is a proper subset of $K_j$, that is, they are
%not equal.

%%FIXME: could we look at the intersection of their respective PPI
%%graphs as a measure of concordance?

%Complete concordance occurs when $C_i = K_j$, such complexes are
%reasonably easy to identify and only one of them is retained. A
%somewhat more difficult case is when $C_i \subset K_j$. For some
%technologies we will retain $C_i$ and for others we want to retain
%$K_j$. The user will have to specify which of the two alternatives
%they want. Both cannot be retained due to identifiability
%problems. %%FIXME: is this true - and do you have an example etc?

%Since much of the data is derived from experimentation and the
%underlying experiments tend to have high error rates (FIXME:citations
%etc here) some consideration of near concordance must also be
%considered. 

%What measures will we use? What graphics/summary statistics can be produced?


\section{The estimate}

The result of combining data from multiple database sources is an aggregate 
protein complex membership \textit{in silico interactome}. The elements of 
this interactome consists of $1755$ proteins and $743$ protein complexes. The 
number of proteins is consistant with biological estimates (%%~40% of genome?)
of the number of expressed genes for the a particular genome;
the number of unique protein complexes gives us enough complexity to 
conduct \textit{in silico} experiments in order to emulate true wet-lab
experimnets.

Histogram for the cardinalities or protein complexes?
Histogram for the order of each particular protein's complex membership?

What else should we put here???
   


\section{Discussion}

What did we learn,

\end{document}
